app/models/school_can_order_devices_notifications.rb

```diff
@@ -1,108 +1,110 @@
-
-    if status?(nil, 'needs_contact', school: school) && user.in?(school.responsible_body.users)
-      :nudge_rb_to_add_school_contact
-    elsif status?('needs_info', 'school_contacted', school: school) && user.in?(school.organisation_users)
-      :user_can_order_but_action_needed
-    elsif status?('rb_can_order', 'school_can_order', school: school) && user.orders_devices? && !user.seen_privacy_notice?
-      :nudge_user_to_read_privacy_policy
-    elsif status?('rb_can_order', school: school) && school.responsible_body.vcap? && user.in?(school.order_users_with_active_techsource_accounts)
-      if school.can_order_routers_only_right_now?
-        :user_can_order_routers_in_virtual_cap
-      else
-        :user_can_order_in_virtual_cap
-      end
-    elsif status?('ready', 'school_ready', 'rb_can_order', 'school_can_order', school: school) && user.in?(school.order_users_with_active_techsource_accounts)
-      if school.responsible_body.new_fe_wave?
-        if school.can_order_routers_only_right_now?
-          :user_can_order_routers_in_fe_college
-        else
-          :user_can_order_in_fe_college
-        end
-      elsif school.can_order_routers_only_right_now?
-        :user_can_order_routers
-      else
-        :user_can_order
-      end
-    end
-  end
-
-  def notify_computacenter_by_email
-    ComputacenterMailer
-      .with(school: school, new_cap_value: new_cap_value)
-      .notify_of_school_can_order
-      .deliver_later
+  def notify_users
+    relevant_users ? notify_about_school_being_able_to_order : notify_support
   end
 
-  def notify_users(users:, school:, message_type:)
-    users.each do |user|
-      notify_user(user: user, school: school, message_type: message_type)
-    end
+  def notify_user(user)
+    message_type = message_for(user) unless school.opted_out?
+    return if message_type.blank?
+
+    CanOrderDevicesMailer.with(user: user, school: school).send(message_type).deliver_later
+    EventNotificationsService.broadcast(UserCanOrderEvent.new(user: user, school: school, type: message_type))
   end
 
-  def notify_user(user:, school:, message_type:)
-    CanOrderDevicesMailer
-      .with(user: user, school: school)
-      .send(message_type)
-      .deliver_later
-    EventNotificationsService.broadcast(
-      UserCanOrderEvent.new(user: user, school: school, type: message_type),
-    )
+  def nudgeable_user?(user)
+    user.orders_devices? && !user.seen_privacy_notice?
   end
 
-  def new_cap_value
-    school&.raw_cap(:laptop)
+  def ordering_user?(user)
+    user.in?(school.order_users_with_active_techsource_accounts)
+  end
+
+  def org_user?(user)
+    user.in?(school.organisation_users)
+  end
+
+  def rb_user?(user)
+    user.in?(school.responsible_body_users)
+  end
+
+  def relevant_users
+    return @relevant_users if instance_variable_defined?(:@relevant_users)
+
+    @relevant_users = school.organisation_users.presence || school.responsible_body_users.presence
```

Comment:
Can this be written as:
`@relevant_users ||= school.organisation_users.presence || school.responsible_body_users.presence`

Or am I missing something?

Reply:
The thing is that if I do what you are suggesting, if the result is nil, everytime this method is called it will trigger the queries and recompute it all again and again. 
I mean the memoization doesn't work if the result is nil.

Reply:
Makes sense: `nil` is a valid value.